File: PequliumWidget/TodayViewController.m
//
//  TodayViewController.m
//  PequliumWidget
//
//  Created by Kyrylo Matvieiev on 22/02/2017.
//  Copyright © 2017 Kyrylo Matvieiev. All rights reserved.
//

#import "TodayViewController.h"
#import <NotificationCenter/NotificationCenter.h>
#import "Manager.h"

@interface TodayViewController () <NCWidgetProviding>
@property (strong, nonatomic) Manager *manager;
@end

@implementation TodayViewController

- (void)viewWillAppear:(BOOL)animated {
    [super viewWillAppear:animated];
}

- (void)viewDidLoad {
    [super viewDidLoad];
    
    self.manager = [Manager sharedInstance];
    
    
    
    NSLog(@"%.2f", [self.manager getMonthDebit]);
    
}

- (void)didReceiveMemoryWarning {
    [super didReceiveMemoryWarning];
    // Dispose of any resources that can be recreated.
}

- (void)widgetPerformUpdateWithCompletionHandler:(void (^)(NCUpdateResult))completionHandler {
    // Perform any setup necessary in order to update the view.
    
    // If an error is encountered, use NCUpdateResultFailed
    // If there's no update required, use NCUpdateResultNoData
    // If there's an update, use NCUpdateResultNewData
    

    
    completionHandler(NCUpdateResultNewData);
}

@end

File: PequliumProject/Pequlium/PequliumWidget/TodayViewController.swift
//
//  TodayViewController.swift
//  PequliumWidget
//
//  Created by Kyrylo Matvieiev on 28/02/2017.
//  Copyright © 2017 Kyrylo Matvieiev. All rights reserved.
//

import UIKit
import NotificationCenter

class TodayViewController: UIViewController, NCWidgetProviding {
    
    private let sharedUserDefaults = UserDefaults(suiteName: "group.pequlium.data")
    private let manager = Manager.sharedInstance
    
    override func viewDidLoad() {
        super.viewDidLoad()
        
    }
    
    override func didReceiveMemoryWarning() {
        super.didReceiveMemoryWarning()
        // Dispose of any resources that can be recreated.
    }
    
    func widgetPerformUpdate(completionHandler: (@escaping (NCUpdateResult) -> Void)) {
        // Perform any setup necessary in order to update the view.
        
        // If an error is encountered, use NCUpdateResult.Failed
        // If there's no update required, use NCUpdateResult.NoData
        // If there's an update, use NCUpdateResult.NewData
        
        completionHandler(NCUpdateResult.newData)
    }
    
}
